service/matching/taskQueueManager.go

```diff
@@ -630,3 +700,71 @@ func newIOContext() (context.Context, context.CancelFunc) {
 
 	return ctx, cancel
 }
+
+func (c *taskQueueManagerImpl) fetchMetadataFromRootPartitionOnInit(ctx context.Context) {
+	if c.metadataInitialFetchFut.Ready() {
```

Comment:
this can't possibly be true, can it?

Reply:
No, but, it could before some re-shuffling. I like the idempotency anyway, no harm in protecting here I think.